browser/ui/webui/brave_rewards_ui.cc

```diff
@@ -4,12 +4,76 @@
 
 #include "brave/browser/ui/webui/brave_rewards_ui.h"
 
+#include "brave/browser/payments/payments_service.h"
+#include "brave/browser/payments/payments_service_factory.h"
 #include "brave/common/webui_url_constants.h"
+#include "chrome/browser/profiles/profile.h"
 #include "components/grit/brave_components_resources.h"
+#include "content/public/browser/web_ui_message_handler.h"
+#include "content/public/common/bindings_policy.h"
+
+
+using content::WebUIMessageHandler;
+
+namespace {
+
+// The handler for Javascript messages for Brave about: pages
+class RewardsDOMHandler : public WebUIMessageHandler {
+ public:
+  RewardsDOMHandler() {
+  }
+  ~RewardsDOMHandler() override {}
+
+  void Init();
+
+  // WebUIMessageHandler implementation.
+  void RegisterMessages() override;
+
+ private:
+  void HandleCreateWalletRequested(const base::ListValue* args);
+  void OnWalletCreated();
+  DISALLOW_COPY_AND_ASSIGN(RewardsDOMHandler);
+};
+
+void RewardsDOMHandler::RegisterMessages() {
+  web_ui()->RegisterMessageCallback("createWalletRequested",
+      base::BindRepeating(&RewardsDOMHandler::HandleCreateWalletRequested,
+                          base::Unretained(this)));
+}
+
+void RewardsDOMHandler::Init() {
+}
+
+void RewardsDOMHandler::HandleCreateWalletRequested(const base::ListValue* args) {
+  Profile* profile = Profile::FromWebUI(web_ui());
+  payments::PaymentsService* payments_service =
+      PaymentsServiceFactory::GetForProfile(profile);
+
+  if (payments_service) {
+    payments_service->CreateWallet();
+  }
+
+  // TODO(bbondy): Use an observer or client override for when the wallet is actually
+  // created once the native-ledger library supports it.
+  OnWalletCreated();
```

Comment:
I know this is temporary, but seems like we should have some kind of message if payments_service is null (windows)?

Reply:
Added `OnWalletCreateFailed`